src/openms/include/OpenMS/CHEMISTRY/TheoreticalSpectrumGeneratorXLMS.h

```diff
@@ -57,6 +58,23 @@ namespace OpenMS
   {
     public:
 
+      struct LossMass
+      {
+        String name;
+        double mass;
+
+        bool operator==(const LossMass & other) const
+        {
+          return name == other.name && mass == other.mass;
+        }
+
+        // even if there are multiple losses with the same mass, having every mass once is enough
+        bool operator<(const LossMass & other) const
+        {
+          return mass < other.mass;
```

Comment:
you need to also compare the name to get a proper less operator
std::tie(mass, name) < std::tie(other.mass, other.name)

Reply:
I agree in general, but I used this only to make a unique list of loss masses. What should I do if two losses have the same mass but different names? Apply both and have two theoretical peaks stacked at the same mass? But I just noticed that this is not really used currently. The losses are now restricted to H2O and NH3, so there won't be any equal masses anyway. I will remove this function for now.

Reply:
in this case it should not be an operator<.
Not providing a full comparison operator actually leads to very nasty bugs in some cases (you can imagine that some algorithms, also in the STL, only use < to determine if two elements are exactly equal)